drivers/hmis/app/models/hmis/hud/processors/geolocation_processor.rb

```diff
@@ -0,0 +1,46 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Hmis::Hud::Processors
+  class GeolocationProcessor < Base
+    def factory_name
+      :clh_location_factory
+    end
+
+    def schema
+      nil
+    end
+
+    def process(field, value)
+      attribute_name = ar_attribute_name(field)
+
+      if attribute_name == 'coordinates'
+        attribute_value = JSON.parse(value, symbolize_names: true)
+        latitude = attribute_value[:latitude]
+        longitude = attribute_value[:longitude]
+        not_collected_reason = attribute_value[:notCollectedReason]
+        raise ArgumentError, 'Geolocation coordinates in unexpected format' unless (latitude && longitude) || not_collected_reason
+
+        return @processor.send(factory_name).destroy if not_collected_reason
+
+        @processor.send(factory_name).assign_attributes(lat: latitude, lon: longitude)
+      else
+        @processor.send(factory_name).assign_attributes(attribute_name => value)
+      end
+    end
+
+    def assign_metadata
+      clh = @processor.send(factory_name, create: false)
+      return if clh&.destroyed?
+
+      clh&.assign_attributes(
+        source: @processor.enrollment_factory,
```

Comment:
untested, comment for linking to other issue..
```
@processor.enrollment_factory.as_warehouse_unpersisted

...

def as_warehouse_unpersisted
   raise 'bad' if persisted?
   GrdaWarehouse::Hud::Enrollment.new(self.attributes)
end
```

Reply:
Here's one idea that maybe works but feels pretty hacky to me: https://github.com/greenriver/hmis-warehouse/pull/4680/files#diff-81d5acd9511ddd9bf95cd056ce5b5252916e30476d6fdb1b3315c85e7d9a4616R15-R21

Reply:
Oh well, that makes sense. I don't mind the after_save solution..